src/TwirpCurlButton.tsx

```diff
@@ -0,0 +1,211 @@
+import React from 'react';
+import {
+  IconButton,
+  Tooltip,
+  Snackbar,
+  Modal,
+  Paper,
+  makeStyles,
+  Table,
+  TableHead,
+  TableRow,
+  TableCell,
+  TableBody,
+  Fade,
+  TextField,
+} from '@material-ui/core';
+import FileCopyIcon from '@material-ui/icons/FileCopy';
+import CloseIcon from '@material-ui/icons/Close';
+import CodeIcon from '@material-ui/icons/Code';
+import { twirpCurl } from './twirp';
+
+const useStyles = makeStyles(theme => ({
+  modal: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'center',
+  },
+  modalContent: {
+    maxWidth: '54em',
+    '&:focus': {
+      outline: 'none',
+    },
+  },
+  title: {
+    display: 'flex',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+    fontSize: '16px',
+    backgroundColor: theme.palette.grey[100],
+    borderTopLeftRadius: theme.shape.borderRadius,
+    borderTopRightRadius: theme.shape.borderRadius,
+  },
+  input: {
+    marginRight: theme.spacing(2),
+  },
+  pre: {
+    margin: 0,
+  },
+  code: {
+    display: 'block',
+    whiteSpace: 'pre-wrap',
+    maxHeight: '54em',
+    overflowY: 'auto',
+    padding: theme.spacing(2),
+    backgroundColor: theme.palette.grey[900],
+    color: 'white',
+    borderRadius: theme.shape.borderRadius,
+    fontSize: '12px',
+    fontFamily: 'Monaco, monospace',
+  },
+  tableCellNoBottomBorder: {
+    borderBottom: 'none',
+  },
+  tableCellNoBottomPadding: {
+    paddingBottom: 0,
+  },
+}));
+
+interface Config {
+  baseUrl: string;
+  username: string;
+  password: string;
+}
+
+const TwirpCurlButton: React.FC<{ method: protobuf.Method }> = ({ method }) => {
```

Comment:
I removed all `React.FC` calls everywhere and switched to named functions.  Just because I don't think it was adding much and gives better readability of the component stack in the browser.